CIP-?/README.md

```diff
@@ -0,0 +1,783 @@
+```
+Perform a bitwise logical AND on arguments of the same
+length, producing a result of the same length, erroring otherwise.
+
+---
+```haskell
+iorByteString :: BuiltinByteString -> BuiltinByteString -> BuiltinByteString
+```
+Perform a bitwise logical IOR on arguments of the same
+length, producing a result of the same length, erroring otherwise.
+
+---
+```haskell
+xorByteString :: BuiltinByteString -> BuiltinByteString -> BuiltinByteString
+```
+Perform a bitwise logical XOR on arguments of the same
+length, producing a result of the same length, erroring otherwise.
+
+---
+```haskell
+complementByteString :: BuiltinByteString -> BuiltinByteString
+```
+Complement all the bits in the argument, producing a
+result of the same length.
+
+---
+Lastly, we define the following additional operations:
+
+```haskell
+shiftByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinByteString
+```
+Performs a bitwise shift of the first argument by a number of bit positions
+equal to the absolute value of the second argument, the direction of the shift
+being indicated by the sign of the second argument.
+
+---
+```haskell
+rotateByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinByteString
+```
+Performs a bitwise rotation of the first argument by a number of bit positions
+equal to the absolute value of the second argument, the direction being
+indicated by the sign of the second argument.
+
+---
+```haskell
+popCountByteString :: BuiltinByteString -> BuiltinInteger
+```
+Returns the number of $1$ bits in the argument.
+
+---
+```haskell
+testBitByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinBool
+```
+If the position given by the second argument is not in
+bounds for the first argument, error; otherwise, if the bit given by that
+position is $1$, return `True`, and `False` otherwise.
+
+---
+```haskell
+writeBitByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinBool -> BuiltinByteString
+```
+If the position given by the second argument is not in bound for the first 
+argument, error; otherwise, set the bit given by that position to $1$ if the 
+third argument is `True`, and $0$ otherwise.
+
+---
+```haskell
+findFirstSetByteString :: BuiltinByteString -> BuiltinInteger
+```
+Return the lowest index such that `testBitByteString` with the first argument 
+and that index would be `True`. If no such index exists, return $-1$ instead.
+
+## Semantics
+
+### Preliminaries
+
+We define $\mathbb{N}^{+} = \\{ x \in \mathbb{N} \mid x \neq 0 \\}$. We assume
+that `BuiltinInteger` is a faithful representation of $\mathbb{Z}$. A
+*bit sequence* $s = s_n s_{n-1} \ldots s_0$ is a sequence such that for
+all $i \in \\{ 0,1,\ldots,n \\}$, $s_i \in \\{ 0, 1 \\}$. A bit sequence 
+$s = s_n s_{n-1} \ldots s_0$ is a *byte sequence* if $n = 8k - 1$ for some 
+$k \in \mathbb{N}$. We denote the *empty bit sequence* (and, indeed, empty byte 
+sequence as well) by $\emptyset$.
```

Comment:
How about this:
"...by $\emptyset$. We do not need to make any assumptions about how bytestrings are actually represented in Plutus, we give the semantics of the operations here, they can be implemented on whatever representation is actually used."

Reply:
I've gone for a slightly different rephrasing: the ordering is still specified, but a little more clearly.